[Le Yew] iP

itsme-zeix opened the pull request:
## Dipsy
> meowlo - The Almighty Dipsy, also my cat and my overlord

PurrfessorDipsy frees your mind of having to remember things you need to do, it's:
* GUI-based (well, eventually)
* Easy to learn
* Crafted to make sure you feed Dipsy more Ciao Churu [(buy here)](https://shopee.sg/-Promo-Buy-4-For-20.65-Each-Ciao-Churu-48s-50sFestive-Packs-i.240182006.8014116763?sp_atk=3911fa4f-85fa-4962-b3ed-9d6376279876&xptdk=3911fa4f-85fa-4962-b3ed-9d6376279876)

All you need to do is:
1. Buy more Ciao Churu
2. Feed Dipsy more Ciao Churu

Dipsy tell you what to do after that happens 😽

Now here is what I need to do to clear the mandatory `markdown` requirements:
- [x] ~~meow~~ **MEOW**
- [ ] nice GUI
- [ ] live laugh love-ing
- [x] will to live

---
Don't we all love some Java code block that is irrelevant to the project?
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

itsme-zeix committed:
Rename main class

itsme-zeix committed:
Add greeting and exit message

itsme-zeix committed:
Add command to echo input and exit program

itsme-zeix committed:
Add memory array/field' to store strings

itsme-zeix committed:
Add save to memory function

itsme-zeix committed:
Add list memory function and command

itsme-zeix committed:
Add message prepended to task list

itsme-zeix committed:
Refactor ArrayList to HashMap

itsme-zeix committed:
refactor: change generic variable names to be more accurate

itsme-zeix committed:
Refactor to use Task class and add mark as done feature

itsme-zeix committed:
Add getStatusIcon method to Task class and refactor other methods to use it

itsme-zeix committed:
Refactor Task class into a separate file

itsme-zeix committed:
Add protected access modifier to fields in Task

itsme-zeix committed:
Refactor name of variable taskName to description in Task

itsme-zeix committed:
Add ToDo class

itsme-zeix committed:
Refactor constructor to remove isDone field

itsme-zeix committed:
Update usage of Task class constructor

itsme-zeix committed:
Add Deadline class

itsme-zeix committed:
Add Event class

itsme-zeix committed:
Added implementation and matchers of ToDo, Deadline and Event in Main

itsme-zeix committed:
Added initialization test

itsme-zeix committed:
Add method to exit program gracefully

itsme-zeix committed:
Remove unused trailing whitespace

itsme-zeix committed:
Add input and expected output for test

itsme-zeix committed:
Add error catching, error messages and usage hints

itsme-zeix committed:
Add horizontal line for error message

itsme-zeix committed:
Update test input/expected with incorrect usage of commands

itsme-zeix committed:
Add handling for unrecognized command and usage hints

itsme-zeix committed:
Move regex patterns to class field

itsme-zeix committed:
Refactored main class to reuse functions

itsme-zeix committed:
Update test input and expected output

itsme-zeix committed:
Refactor: removed some redundant methods

itsme-zeix committed:
Refactor main class to merge task creation logic into a factory method

itsme-zeix committed:
Fixed mistake that repeated 'Error:'

itsme-zeix committed:
Add Exception class and implement in main class

itsme-zeix committed:
Refactor: Convert array to ArrayList

itsme-zeix committed:
Add delete function and error catching

itsme-zeix committed:
Add invalid delete error catching into Exception and update naming of invalid mark command

itsme-zeix committed:
Updated test input/expected file to include testing for delete

itsme-zeix committed:
Remove unused Collection package

itsme-zeix committed:
Refactored various Tasks into a Task package

itsme-zeix committed:
Refactored Exceptions into Exception package

itsme-zeix committed:
Update imports in main file

itsme-zeix committed:
Refactor: Change Task to an abstract class and add formatForStorage abstract method

itsme-zeix committed:
Revert "Refactor: Change Task to an abstract class and add formatForStorage abstract method"

This reverts commit 5953996c558b850c7e7ecc3aaf643f372cf57cfa.

itsme-zeix committed:
Add formatForCSV method to Task and children classes

itsme-zeix committed:
Refactor method name formatForCSV to formatToCSV

itsme-zeix committed:
Refactor method name formatForCSV to formatToCSV

itsme-zeix committed:
Add utility to save tasks to file in Storage

itsme-zeix committed:
Add logic to save task list after each action into main

itsme-zeix committed:
Change file created from text to csv format

itsme-zeix committed:
fix saving of description and add getTaskType method

itsme-zeix committed:
Add overloaded constructor for initializing Task with isDone parameter

itsme-zeix committed:
Add loading of task from local storage

itsme-zeix committed:
Merge branch 'branch-Level-7'

itsme-zeix committed:
Update debug statements

itsme-zeix committed:
Remove empty task file log statement

itsme-zeix committed:
Polish printed text

itsme-zeix committed:
Remove ascii art in print

itsme-zeix committed:
Add ASCII art and remove extra space

itsme-zeix committed:
Update test expected output

itsme-zeix committed:
Merge branch 'branch-Level-7'

itsme-zeix committed:
Refactor types for dates from String to LocalDate in Deadline and Event classes

itsme-zeix committed:
Refactor types for initialization of Deadline and Event objects in Storage class

itsme-zeix committed:
Refactor types for initialization of Deadline and Event objects in main class

itsme-zeix committed:
Add DateTime parsing to toString method for Deadline and Event class

itsme-zeix committed:
Refactor Storage.java: remove log for successful save to task file

itsme-zeix committed:
Add method to get dates in Deadline and Event classes

itsme-zeix committed:
Add method to get tasks by date in main class

itsme-zeix committed:
Fix: change parsing for list to allow for addition of date in main class

itsme-zeix committed:
Add format date for storage feature

itsme-zeix committed:
Chore: update usage description with date format and list <date> command

itsme-zeix committed:
Add UnknownCommandException

itsme-zeix committed:
Refactor PurrfessorDipsyException to InvalidCommandException

itsme-zeix committed:
Refactor updated exceptions into main class

itsme-zeix committed:
Chore: remove extra new line from usage hint

itsme-zeix committed:
Refactor delete method in main class

itsme-zeix committed:
Refactor task creation methods for better readability

itsme-zeix committed:
Refactor relevant date retrieval method in Task and listing of tasks on relevant in main class

itsme-zeix committed:
Update tests

itsme-zeix committed:
Fix merge conflicts

itsme-zeix committed:
Fix broken test caused by incorrect merge conflict fix

itsme-zeix committed:
Refactored Ui out of Main

itsme-zeix committed:
Refactored TaskList out of Main

itsme-zeix committed:
Add load method for Storage

itsme-zeix committed:
Update storage to be static

itsme-zeix committed:
Move Storage class into Storage package

itsme-zeix committed:
Add all files into package

itsme-zeix committed:
Refactor classes to adhere to OOP principles and shift into packages for ease of access

itsme-zeix committed:
Fix file compilation logic to consider packages in runtest script

itsme-zeix committed:
Merge branch 'A-MoreOOP-Packages' without fast forward

itsme-zeix committed:
Merge branch 'add-gradle-support'

itsme-zeix committed:
Refactor DateTimeParser to DateParser

itsme-zeix committed:
Fix: add logic for date validation due to java date normalization

itsme-zeix committed:
Add unit tests for ListCommand class

itsme-zeix committed:
Add unit tests for DateParser class

itsme-zeix committed:
Merge branch 'Add-JUnit-Tests' without fast forward

itsme-zeix committed:
Chore: Add JavaDoc to classes in Command package

itsme-zeix committed:
Chore: Add JavaDoc to classes in Exception package

itsme-zeix committed:
Chore: Add JavaDoc to classes in Parser package

itsme-zeix committed:
Chore: Add JavaDoc to classes in Storage package

itsme-zeix committed:
Chore: Add JavaDoc to classes in Task package

itsme-zeix committed:
Chore: Add JavaDoc to classes in TaskList package

itsme-zeix committed:
Chore: Add JavaDoc to classes in Ui package

itsme-zeix committed:
Chore: Add JavaDoc to the PurrfessorDipsy class (main class)

itsme-zeix committed:
Merge branch 'A-JavaDoc'

itsme-zeix committed:
Fix: add storage load on boot that was previously removed during a refactor

itsme-zeix committed:
Fix: add missing logic to load tasks from storage (previously removed during refactor)

itsme-zeix committed:
Refactor: rename packages from PascalCase to lowercase

itsme-zeix committed:
Add method to check for presence of keyword in description in Task class

itsme-zeix committed:
Add method to filter list of task based on keyword in TaskList class

itsme-zeix committed:
Refactor name of method to filter list of task by keyword in TaskList class

itsme-zeix committed:
Add invalid find command error type and usage hint for InvalidCommandException

itsme-zeix committed:
Add FindCommand class

itsme-zeix committed:
Add method for printing tasks that match keyword in Ui and FindCommand

itsme-zeix committed:
Add find command into parser

itsme-zeix committed:
Fix: change order of operations of matcher to prevent getting keyword matched before matcher matches

itsme-zeix committed:
Add exact keyword matching logic into Task class

itsme-zeix committed:
Resolve merge conflict

itsme-zeix committed:
Merge branch 'CodingStandard'

itsme-zeix committed:
Update JUnit package names

itsme-zeix committed:
Fix: fix mispelling of package name

itsme-zeix committed:
Chore: remove unused import in tests

itsme-zeix committed:
Rename PurrfessorDipsy package to lowercase

itsme-zeix committed:
Rename test methods and variables

itsme-zeix committed:
Fix mainClass attribute in gradle config

itsme-zeix committed:
Fix archiveBaseName attribute in gradle config

itsme-zeix committed:
Fix package path in mainClass attribute

Darren-Tung commented on src/main/java/purrfessordipsy/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import purrfessordipsy.exception.InvalidCommandException;
```
You may consider separating the imports between lines 6 and 7, to separate the java packages from the rest

Darren-Tung commented on src/main/java/purrfessordipsy/command/EventCommand.java:
```diff
@@ -0,0 +1,62 @@
+import purrfessordipsy.parser.DateParser;
+import purrfessordipsy.task.Event;
+import purrfessordipsy.tasklist.TaskList;
+import purrfessordipsy.ui.Ui;
```
Likewise, to the previous preview, you can consider leaving a gap between lines 8 and 9 to separate the imports.

Darren-Tung commented on src/main/java/purrfessordipsy/command/ListCommand.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    private ArrayList<Task> filterTasksByDate(LocalDate date) {
+        ArrayList<Task> res = new ArrayList<>();
```
You may want to consider using a better variable name instead of the name 'res'

yyanyee replied:
i agree. variable name `res` is unclear

Darren-Tung commented on src/main/java/purrfessordipsy/exception/InvalidCommandException.java:
```diff
@@ -0,0 +1,61 @@
+            case INVALID_FIND_COMMAND ->
+                    "'find' command requires a keyword to search for.\nUsage: find <keyword>";
+            default -> "Unknown invalid command. Please try again." ;
+        };
```
Do note that your switch case should be in a different style.

Darren-Tung commented on src/main/java/purrfessordipsy/parser/Parser.java:
```diff
@@ -0,0 +1,25 @@
+            case "find" -> new FindCommand(userInput, tasks, ui);
+            case "bye" -> new ByeCommand(userInput, tasks, ui);
+            default -> throw new UnknownCommandException();
+        };
```
Similar to my previous comment, you can consider changing your switch case style.

Darren-Tung reviewed (commented):
Overall, I found your code to be well written and easy to read. I have read the first few files and so far, it just requires some minor changes mostly, coding standard violations.

Good work 👍

yyanyee commented on src/main/java/purrfessordipsy/command/MarkCommand.java:
```diff
@@ -0,0 +1,83 @@
+        Matcher markMatcher = MARK_PATTERN.matcher(super.userInput);
+        if (markMatcher.matches()) {
+            String action = markMatcher.group(1);
+            int userGivenIndex = Integer.parseInt(markMatcher.group(2));
```
`userGivenIndex` can be shortened to eg `givenIndex` etc

yyanyee commented on src/main/java/purrfessordipsy/storage/Storage.java:
```diff
@@ -0,0 +1,179 @@
+     * @return A string representing the tasks in CSV format.
+     */
+    private static String formatTasks(ArrayList<Task> tasks) {
+        StringBuilder res = new StringBuilder();
```
`res` name has unclear meaning

yyanyee commented on src/main/java/purrfessordipsy/storage/Storage.java:
```diff
@@ -0,0 +1,179 @@
+        try {
+            File parentDir = TASK_FILE.getParentFile();
+            if (!parentDir.exists()) {
+                boolean success = parentDir.mkdirs();
```
use clearer naming instead of `success`

yyanyee commented on src/main/java/purrfessordipsy/task/Event.java:
```diff
@@ -0,0 +1,94 @@
+     */
+    @Override
+    public String formatToCSV() {
+        String res = super.formatToCSV();
```
as mentioned earlier, you can consider renaming your `res` variable to something clearer

itsme-zeix committed:
Add Checkstyle configs

itsme-zeix committed:
Fix code style

itsme-zeix committed:
Fix code style for Ui class

itsme-zeix committed:
Fix name style for methods in test

itsme-zeix committed:
Merge branch 'Branch-A-Checkstyle'

itsme-zeix committed:
Fix name style for methods in ListCommandTest

itsme-zeix committed:
Merge branch 'Branch-A-CheckStyle'

itsme-zeix committed:
Rename PurrfessorDipsy to Dipsy

itsme-zeix committed:
Add functioning Level-10

itsme-zeix committed:
Add CSS styling from JavaFX tutorial

itsme-zeix committed:
Merge branch 'Branch-Level-10'

itsme-zeix committed:
Update JavaDoc for command package

itsme-zeix committed:
Revert "Update JavaDoc for command package"

This reverts commit 01699b29bc4ed04d8d0707c2c33a197f854362dc.

itsme-zeix committed:
Update JavaDoc for command package

itsme-zeix committed:
Update mainClass in build.gradle

itsme-zeix committed:
Update dependencies in build.gradle

itsme-zeix commented:
> Overall, I found your code to be well written and easy to read. I have read the first few files and so far, it just requires some minor changes mostly, coding standard violations.
> 
> Good work 👍

Thanks for the review Darren! Great catches, all fixed up.

itsme-zeix committed:
Update JavaDoc in Ui and Dipsy class

itsme-zeix committed:
Modify getErrorMessage to use varArgs

yyanyee reviewed (commented):
Overall good! Just some minor cosmetic changes

itsme-zeix committed:
Address code quality comments

itsme-zeix committed:
Add assertions to dipsy package

itsme-zeix committed:
Refactor to improve code quality

itsme-zeix committed:
Add graceful exit for application

A method to shut down the application after a delay was added to
MainWindow. This method is called by Dipsy.

This ensures that the farewell message is shown to the user
before the application is closed.

itsme-zeix committed:
Refactor getTaskByKeyword method to use streams

itsme-zeix committed:
Refactor: remove unused import

itsme-zeix committed:
Merge pull request #1 from itsme-zeix/branch-A-Assertions

Add assertions to dipsy package

itsme-zeix committed:
Merge master into branch-A-CodeQuality

itsme-zeix committed:
Merge pull request #2 from itsme-zeix/branch-A-CodeQuality

Improve code quality and add graceful application exit

itsme-zeix committed:
Merge branch 'master' into branch-A-Streams

itsme-zeix committed:
Merge pull request #3 from itsme-zeix/branch-A-Streams

Refactor to use streams

itsme-zeix committed:
Fix Java version in FXML config file from 22 to 17

itsme-zeix committed:
Add gradle workflow

itsme-zeix committed:
Remove deprecated IO redirection tests in gradle config

itsme-zeix committed:
Fix tests

itsme-zeix committed:
Fix ListCommandTest

itsme-zeix committed:
Merge branch 'branch-A-CI'

itsme-zeix committed:
Merge branch 'master' into branch-A-CodeQuality

itsme-zeix committed:
Refactor Ui class

itsme-zeix committed:
Refactor MarkCommand class

itsme-zeix committed:
Merge branch 'branch-A-CodeQuality'

itsme-zeix committed:
Fix: missing semicolon in Ui class

itsme-zeix committed:
Refactor Dipsy class

itsme-zeix committed:
Fix checkstyle issues

itsme-zeix committed:
Add help button and modify UI

itsme-zeix committed:
Merge branch 'branch-C-Help'

itsme-zeix committed:
Add hint for text input box

itsme-zeix committed:
Revamp GUI

itsme-zeix committed:
Merge branch 'Branch-A-BetterGUI'

itsme-zeix committed:
Fix CheckStyle errors

itsme-zeix committed:
Add product screenshot

itsme-zeix committed:
Fix index parsing in DeleteCommand

itsme-zeix committed:
Update UnknownCommandException message to redirect to help button

itsme-zeix committed:
Add README User Guide

itsme-zeix committed:
Remove incorrect commands in command summary

itsme-zeix committed:
Merge branch 'branch-A-UserGuide'

itsme-zeix committed:
Add help menu feature

itsme-zeix committed:
Merge branch 'branch-A-UserGuide'

itsme-zeix committed:
Refactor MarkCommand style

itsme-zeix committed:
Add error catching for incorrect index in DeleteCommand

itsme-zeix committed:
Add GitHub repo README

itsme-zeix committed:
Fix screenshot URL for GitHub repo README

itsme-zeix committed:
Refactor README for better organization

itsme-zeix committed:
Refactor README again for better organization

itsme-zeix committed:
Update JAR output file name

itsme-zeix committed:
Add app title and icon

itsme-zeix committed:
Add Quick Start guide

itsme-zeix committed:
Fix formatting of README

itsme-zeix committed:
Fix formatting of list feature in README
